### PR TITLE
Support `ignoredExceptionsForType`

### LINKED ISSUE #1046: Support `ignoredExceptionsForType`
### Description

Similar to https://docs.sentry.io/platforms/android/configuration/options/#ignored-exceptions-for-type
A little bit more user-friendly in case you wanna ignore specific error types rather than `if error is type` checks on `beforeSend` callback.
Consider using the "right" naming, See https://github.com/getsentry/sentry-java/issues/1532

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Support `ignoredExceptionsForType`

## :bulb: Motivation and Context

Closes #1046

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: dart/lib/src/sentry_options.dart [hunk 1]
```diff
@@ -334,6 +334,22 @@ class SentryOptions {
...
+
+  /// Adds exception type to the list of ignored exceptions.
+  void addExceptionFilterForType(Type exceptionType) {
+    _ignoredExceptionsForType.add(exceptionType);
+  }
+
+  /// Check if [ignoredExceptionsForType] contains an exception.
+  bool containsIgnoredExceptionForType(dynamic exception) {
+    return exception != null &&
+        _ignoredExceptionsForType.contains(exception.runtimeType);
```
Comment (buenaflor) [inline dart/lib/src/sentry_options.dart:350]: how is the behaviour on flutter web here? 

I think the runtimetype there is minified? afaik

Comment (denrase) [reply dart/lib/src/sentry_options.dart:350]: Yes, it's minified, but always has the same value during runtime, so the filter works as expected.

Comment (buenaflor) [reply dart/lib/src/sentry_options.dart:350]: got it 👍

File: dart/lib/src/sentry_client.dart [hunk 2]
```diff
@@ -83,6 +83,16 @@ class SentryClient {
...
     Hint? hint,
   }) async {
+    if (_options.containsIgnoredExceptionForType(event.throwable)) {
+      _options.logger(
+        SentryLevel.debug,
+        'Event was dropped as the exception ${event.throwable.runtimeType.toString()} is ignored.',
+      );
+
+      _recordLostEvent(event, DiscardReason.eventProcessor);
+      return _sentryId;
```
Comment (buenaflor) [inline dart/lib/src/sentry_client.dart]: this isn't really part of this PR but I feel like this should be renamed to something like `_emptySentryId`

**Review by buenaflor — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Features
+
+- Support `ignoredExceptionsForType` ([#2150](https://github.com/getsentry/sentry-dart/pull/2150))
```
Comment (buenaflor) [inline CHANGELOG.md:7]: can you give a very quick example how to use it here

**Review by buenaflor — APPROVED** [review]
just one more thing then we're good to go

**Pull request merged**